### ISSUE TITLE
The borders of components within a ControlGroup do not line up

### ISSUE THREAD

**Issue by David-Melo** [palantir/blueprint#4044]
#### Environment

- __Package version(s)__: 3.24.0
- __Browser and OS versions__: MacOS Mojave (10.14.4) - Chrome 80.0.3987.149 (Also In Firefox & Safari)

#### Steps to reproduce
1. None, this issue is evident in the blueprint website & Documentation for the ControlGroup component.

#### Actual behavior
Border or Box Shadow or Inset (Not sure what exactly) Is not lining up between buttons and inputs inside a Control Group. Issue seems to be only in the Dark Theme (See Attached Screenshots)

#### Expected behavior
Border or Box Shadow or Inset (Not sure what exactly) should align between the components within a Control Group

#### Possible solution
None

![image](https://user-images.githubusercontent.com/1429949/78595830-af129400-7818-11ea-8c8f-b5b3a8a4ce58.png)

**Comment by David-Melo** [comment]
I narrowed the issue down to the box shadow css for the **.bp3-dark .bp3-input** class. I made a quick patch for my app, but it's not perfect. Since the box shadow's have some transparency,  even when being set to the same value as the other components, the overlapping transparencies add up and cause the overlaps to be darker.

`.bp3-dark .bp3-input {
    -webkit-box-shadow: 0 0 0 1px rgb(31, 42, 50);
    box-shadow: 
        0 0 0 1px rgba(16,22,26,.4), 
        0 0 0 1px rgba(16,22,26,.4), 
        0 0 0 1px rgba(16,22,26,.4), 
        inset 0 0 0 0 rgba(16,22,26,.3), 
        inset 0 0 0 0 rgba(16,22,26,.4)
}`

**Comment by justinbhopper** [comment]
It seems Buttons and HtmlSelects are using an outer box-shadow in the dark-theme, whereas the light theme uses an inset box-shadow.  It will probably take quite some work to fix all the styling.

**Comment by adidahiya** [comment]
@David-Melo yes this is a minor bug. I don't have an immediate solution but I will tell you that we spent a _lot_ of time early on in Blueprint development to styles the button borders just right in both light & dark theme (probably more time than I would like to admit). The decision to use outer/inner box-shadow was not taken lightly. Yes we can potentially fix this bug but it is tricky to do so while preserving the aesthetics we want. It's also not very high on the priority list right now.
